### PR TITLE
Initial commit of feature-complete v1 version

### PR DESCRIPTION
**Pull request by polypour**
@shrik450 could you help to take a quick look and approve the PR please?  :-)

### CODE REVIEW

File: entrypoint.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        data={"grant_type": "client_credentials"},
+        headers=headers,
+        auth=(client_id, client_secret),
+    )
+    # Populate the access token return value
+    access_token = (
+        response.json()["access_token"] if response.status_code == 200 else None
+    )
+    # Return response status code and access token
+    return (response.status_code, access_token)
```
Comment (shrik450) [inline entrypoint.py:61]: I don't see this response code being used anywhere. Is there a reason we're returning it?

File: entrypoint.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    context = {"bom": bom_items_digikey_data}
+    # Create report output folder if it doesn't exist
+    try:
+        os.makedirs("/component_report")
+    except FileExistsError:
+        pass
+    # Unzip the JS/CSS assets
+    shutil.unpack_archive("/report_template/assets.zip", "/component_report")
+    # Write HTML output file
+    with ExitStack() as stack:
```
Comment (shrik450) [inline entrypoint.py:255]: You don't need an exit stack here, you can directly use:

```py
with open(...) as report_file:
```

File: report_template/assets/bootstrap/css/bootstrap.min.css
Comment (shrik450) [inline report_template/assets/bootstrap/css/bootstrap.min.css:1]: Do we need to keep these files here and in the zip? I think just the zip should be enough?

File: report_template/index.html [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		{% else %}
+        <div style="margin-left: 30px;margin-right: 30px;border-style: solid;border-color: var(--bs-gray-500);border-radius: 3px;padding-bottom: 0px;">
+		{% endif %}
+            <div class="row" style="font-size: 12px;color: rgb(255,255,255);margin-right: 0px;margin-left: 0px;">
+                <div class="col" style="background: var(--bs-secondary-text-emphasis);padding-top: 5px;padding-bottom: 5px;"><span>{{ part.associated_refdes }}</span></div>
+            </div>
+			{% if part.mfr_name is not none %}
+            <div class="row g-0">
+                <div class="col-auto text-center" style="padding-right: 20px;padding-left: 20px;border-style: none;border-color: var(--bs-gray-400);"><img src="{{ part.photo_url }}" style="height: 88px;margin-left: 10px;"><a href="{{ part.datasheet_url }}" target="_blank">
+                        <p style="text-align: center;font-size: 10px;padding-bottom: 5px;"><i class="far fa-file-pdf" style="font-size: 20px;color: var(--bs-form-invalid-color);"></i></p>
```
Comment (shrik450) [inline report_template/index.html:33]: We're only using font awesome for this one icon AFAICT, is it worth adding all of its assets for this?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# DigiKey-Reports
+An actions repository for DigiKey API integration and report generation given an input BOM
```
[the remark this answers is not in the document]
Comment (polypour) [reply README.md:2]: That's a good idea. Let's add that to the README.

File: entrypoint.py [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+DIGIKEY_API_URL_BASE = "https://api.digikey.com"
+DIGIKEY_API_AUTH_ENDPOINT = DIGIKEY_API_URL_BASE + "/v1/oauth2/token"
+DIGIKEY_API_V4_KEYWORD_SEARCH_ENDPOINT = (
+    DIGIKEY_API_URL_BASE + "/products/v4/search/keyword"
+)
+
+
+################################################################################
+class ComponentData:
```
[the remark this answers is not in the document]
Comment (polypour) [reply entrypoint.py:21]: That's a good point, I'll add the decorator.

File: entrypoint.py (hunk 1 above)
Comment (polypour) [reply entrypoint.py:61]: Good catch, the response code is being used in the API query endpoint but not in the auth. I'll add the check to auth to exit if auth fails.

File: report_template/assets/bootstrap/css/bootstrap.min.css
Comment (polypour) [reply report_template/assets/bootstrap/css/bootstrap.min.css:1]: Agree, let's can the assets folders and keep only the zip

File: .github/workflows/test.yml
Comment (danielallspice) [inline .github/workflows/test.yml:1]: Rename to semantic meaning. What is this workflow doing? Name the file that.

Comment (polypour) [reply .github/workflows/test.yml:1]: Good catch. It's named as test.py in generate-bom and cofactr-cogs also - we will prolly want to rename those as well. It's a ruff format/lint check workflow - we should rename it as that.

File: report_template/index.html (hunk 3 above)
Comment (polypour) [reply report_template/index.html:33]: I'm able to replace the font awesome full library with just the SVG I need for that icon - will make the change.

**Comment by polypour** [comment]
Implemented feedback changes, ready to merge.

**Pull request merged**